pyuvdata/uvdata/uvdata.py

```diff
@@ -3732,40 +4062,59 @@ def fast_concat(
                     raise ValueError(msg)
 
         if axis == "freq":
-            this.freq_array = np.concatenate(
-                [this.freq_array] + [obj.freq_array for obj in other], axis=1
-            )
-            this.Nfreqs = sum([this.Nfreqs] + [obj.Nfreqs for obj in other])
+            if this.future_array_shapes:
+                this.freq_array = np.concatenate(
+                    [this.freq_array] + [obj.freq_array for obj in other]
+                )
+                this.channel_width = np.concatenate(
+                    [this.channel_width] + [obj.channel_width for obj in other]
+                )
+                this.Nfreqs = sum([this.Nfreqs] + [obj.Nfreqs for obj in other])
+            else:
```

Comment:
Another place where I think I failed to add handling for the `channel_width` array with `flex_spw=True` -- would be good to add this in there as well (feel free to hand these off to me, since they are my oversights).

Reply:
yes, I just found this one and some others. I'm realizing there's no testing for the add or fast concat (and maybe the select) methods with flexible spws. I can add some, can you point me at some good files to use?

Reply:
I suspect the either the MIR test file (`sma_test.mir`) or the CARMA-SZA test file (`carma_miriad`) would work, since they both have multiple spws in the dataset that you could in principle select on and then add back together. Again, if you want, I'm happy to try to add that in there, since they're effectively leftovers from my PR.

Reply:
Ok, I'd be happy for you to. I think I have all the logic updated properly, I'll push that up. I do want to make sure we're exercising everything so I'd like to see that your new tests fail before my updated logic is applied (so you'd need to check out the commit before my update or just not pull my latest commit at first). In particular, we need to test cases with one flexible spw.